cluster-autoscaler/cloudprovider/externalgrpc/externalgrpc_cloud_provider.go

```diff
@@ -262,6 +264,68 @@ func (e *externalGrpcCloudProvider) GetAvailableGPUTypes() map[string]struct{} {
 	return gpuTypes
 }
 
+// GetNodeAccelerator returns the label and resource name for the accelerator added to node.
+func (e *externalGrpcCloudProvider) GetNodeAccelerator(node *apiv1.Node) cloudprovider.Accelerator {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+
+	nodeID := node.Name + node.Spec.ProviderID //ProviderID is empty in some edge cases
+	// lookup cache
+	if accelerator, ok := e.acceleratorForNodeCache[nodeID]; ok {
+		klog.V(5).Infof("Returning cached information for GetNodeAccelerator for node %v - %v", node.Name, node.Spec.ProviderID)
+		return accelerator
+	}
+	// perform grpc call
+	ctx, cancel := context.WithTimeout(context.Background(), grpcTimeout)
+	defer cancel()
+	klog.V(5).Infof("Performing gRPC call GetNodeAccelerator for node %v - %v", node.Name, node.Spec.ProviderID)
+	res, err := e.client.GetNodeAccelerators(ctx, &protos.GetNodeAcceleratorsRequest{
+		Node: externalGrpcNode(node),
+	})
+	if err != nil {
+		klog.V(1).Infof("Error on gRPC call GetNodeAccelerator: %v", err)
+		return cloudprovider.Accelerator{Label: "", Type: "", ResourceName: ""}
+	}
+	pbAccelerator := res.GetAccelerator()
+	accelerator := cloudprovider.Accelerator{
+		Label:        pbAccelerator.GetLabel(),
+		Type:         pbAccelerator.GetType(),
+		ResourceName: apiv1.ResourceName(pbAccelerator.GetResourceName()),
+	}
+	e.acceleratorForNodeCache[nodeID] = accelerator
+	return accelerator
+}
+
+// GetAvailableAccelerators returns all available accelerators the cloud provider supports.
+func (e *externalGrpcCloudProvider) GetAvailableAccelerators() []cloudprovider.Accelerator {
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+
+	if e.availableAcceleratorsCache != nil {
+		return e.availableAcceleratorsCache
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), grpcTimeout)
+	defer cancel()
+	klog.V(5).Info("Performing gRPC call GetAvailableAccelerators")
+	res, err := e.client.GetAvailableAccelerators(ctx, &protos.GetAvailableAcceleratorsRequest{})
+	if err != nil {
+		klog.V(1).Infof("Error on gRPC call GetAvailableAccelerators: %v", err)
+		return nil
+	}
+	availableAccelerators := []cloudprovider.Accelerator{}
+	for _, pbAccelerator := range res.GetAvailableAccelerators() {
+		accelerator := cloudprovider.Accelerator{
+			Label:        pbAccelerator.GetLabel(),
+			Type:         pbAccelerator.GetType(),
+			ResourceName: apiv1.ResourceName(pbAccelerator.GetResourceName()),
+		}
+		availableAccelerators = append(availableAccelerators, accelerator)
+	}
+	e.availableAcceleratorsCache = availableAccelerators
+	return availableAccelerators
+}
```

Comment:
This separate cache is really not needed, you could just use the `GpuLabel()` and `GetAvailableGPUTypes()` which already have caching.

Reply:
Removed these caches. Now only using `GPULabel()` and `GetAvailableGPUTypes()`